scipy/linalg/__init__.py

```diff
@@ -38,6 +38,7 @@
    tril - Construct a lower-triangular matrix from a given matrix
    triu - Construct an upper-triangular matrix from a given matrix
    orthogonal_procrustes - Solve an orthogonal Procrustes problem
+   balance - Balance matrix entries with a diagonal similarity transformation
```

Comment:
The function itself is known as such due to matlab and other software. But I am open for ideas. Do you have something specific in mind?